modules/porous_flow/src/userobjects/AdvectiveFluxCalculatorBase.C

```diff
@@ -173,6 +173,42 @@ AdvectiveFluxCalculatorBase::timestepSetup()
       buildCommLists();
 
     _resizing_needed = false;
+
+    // Clear all member vectors
+    _dij.clear();
+    _dDij_dKij.clear();
+    _dDij_dKji.clear();
+    _dDii_dKij.clear();
+    _dDii_dKji.clear();
+    _lij.clear();
+    _rP.clear();
+    _rM.clear();
+    _drP.clear();
+    _drM.clear();
+    _drP_dk.clear();
+    _drM_dk.clear();
+    _fa.clear();
+    _dfa.clear();
+    _dFij_dKik.clear();
+    _dFij_dKjk.clear();
+
+    // Size member vectors
+    _dij.resize(_number_of_nodes);
+    _dDij_dKij.resize(_number_of_nodes);
+    _dDij_dKji.resize(_number_of_nodes);
+    _dDii_dKij.resize(_number_of_nodes);
+    _dDii_dKji.resize(_number_of_nodes);
+    _lij.resize(_number_of_nodes);
+    _rP.resize(_number_of_nodes);
+    _rM.resize(_number_of_nodes);
+    _drP.resize(_number_of_nodes);
+    _drM.resize(_number_of_nodes);
+    _drP_dk.resize(_number_of_nodes);
+    _drM_dk.resize(_number_of_nodes);
+    _fa.resize(_number_of_nodes);
+    _dfa.resize(_number_of_nodes);
+    _dFij_dKik.resize(_number_of_nodes);
+    _dFij_dKjk.resize(_number_of_nodes);
```

Comment:
If you don't care about the content of the vectors you can drop all the `.clear()` calls, as resize will work in those cases. If you do care about the content you can use `.assign()` and still drop the `.clear()` calls. This is fine though.